protocol-units/da-sequencer/node/src/storage/mod.rs

```diff
@@ -188,7 +191,84 @@ impl DaSequencerStorage for Storage {
 	}
 
 	fn produce_next_block(&self) -> Result<Option<SequencerBlock>, DaSequencerError> {
-		todo!()
+		let cf_pending = self.db.cf_handle(cf::PENDING_TRANSACTIONS).ok_or_else(|| {
+			DaSequencerError::StorageAccess("Missing colomn family: pending transactions".into())
+		})?;
+
+		// Collect and deserialize all pending transactions
+		let mut transactions: Vec<Transaction> = Vec::new();
+
+		let iter = self.db.iterator_cf(&cf_pending, rocksdb::IteratorMode::Start);
+		for item in iter {
+			let (_key, value) = item.map_err(|e| DaSequencerError::RocksDbError(e.to_string()))?;
+			let tx: Transaction = bcs::from_bytes(&value)
+				.map_err(|e| DaSequencerError::Deserialization(e.to_string()))?;
+			transactions.push(tx);
+		}
+
+		if transactions.is_empty() {
+			return Ok(None);
+		}
+
+		// Sort transactions using their Ord implementation
+		transactions.sort();
```

Comment:
You can't use the natural order of the movement_type::Transaction because it's not the order we need. Natural order is used on the fullnode size to build the batch. Here we need to order by timestamp first, then the batch order.
I think you need to add the batch timestamp to the Tx storage.

Reply:
What do you mean by the batch order? The order at which the tx's were received?

Reply:
In the #1164 issue, line 2 the order of Tx are `order the Tx by batch timestamp / tx order in batch` .
The Tx are ordered correctly by the fullnode in the batch, but not between batch. So first you need to order by batch (ie: batch timestamp, the oldest first) then the order in the batch.

Reply:
I've used Composite key to store in Rocksdb
```rust
#[derive(Debug, Clone, PartialEq, Eq)]
struct TxCompositeKey {
	timestamp: u64,
	index_in_batch: u32,
	tx_id: [u8; 32],
}
```
This means that rocksdb natural ordering will be lexical (when using big endian encoding) and should be sufficient, also I added the test 
`test_write_batch_naturally_orders_lexicographically_with_composite_key`

Which shows that rocksdb correctly orders on timestamp even if written at different times. Let me know if think it's sufficient for our needs.

Reply:
I've used Composite key to store in Rocksdb
```rust
#[derive(Debug, Clone, PartialEq, Eq)]
struct TxCompositeKey {
	timestamp: u64,
	index_in_batch: u32,
	tx_id: [u8; 32],
}
```
This means that rocksdb natural ordering will be lexical (when using big endian encoding) and should be sufficient, also I added the test 
`test_write_batch_naturally_orders_lexicographically_with_composite_key`

Which shows that rocksdb correctly orders on timestamp even if written at different times. Let me know if think it's sufficient for our needs.

Reply:
Ok yes I didn't see your remarks. Yes it should be ok. I think we must add some test about it. We'll open a new PR for the test.